ocaml/xapi/xapi_pool.ml

```diff
@@ -3698,3 +3698,51 @@ let set_local_auth_max_threads ~__context:_ ~self:_ ~value =
 
 let set_ext_auth_max_threads ~__context:_ ~self:_ ~value =
   Xapi_session.set_ext_auth_max_threads value
+
+let get_auth_filename tmp_dir tarpath =
+  let filename =
+    if String.contains tarpath '/' then
+      Filename.basename tarpath
+    else
+      tarpath
+  in
+  let tmp_auth_file = Filename.concat tmp_dir filename in
```

Comment:
This is risky because there could be clashes over this path if `tmp_dir` is some public temp dir.

Reply:
https://ocaml.org/manual/5.1/api/Filename.html#VALtemp_file should be used here